### ISSUE TITLE
Solved: Ruby 1.8 program locale issue, reload the puts method

### ISSUE THREAD

**Issue by mhtvsSFrpHdE** [tako774/tsk_report#8]
I'm working on UTF-8 and multiple languages support these days.

I see this file:
```
#https://github.com/tako774/tsk_report/blob/master/lib/tenco_report/stdout_to_cp932_converter.rb

# coding: utf-8

require 'nkf'

# convert $stdout to cp932
if !defined?($stdout._write)
  class << $stdout
    alias :_write :write
    def write(str)
      _write NKF.nkf('-sxm0 --cp932', str.to_s)
    end
  end
end
```

And line 22 in skr_report.rb:
```
#https://github.com/tako774/tsk_report/blob/master/skr_report.rb

require 'tenco_report/stdout_to_cp932_converter'
```

But why?

I guess that if I don't understand anything, this line would be the key part:
```
_write NKF.nkf('-sxm0 --cp932', str.to_s)
```

I know that cp932 is the code page of Japanese locale, and what about "smx0"?

If I want it to output something like cp1252 or cp936, just replace "--cp932".

But I'm not sure the "-sxm0".

Do you have some document or tutorial about such option?

And, an extra question.

I want to convert to local ANSI characters set on Windows XP system. But don't convert output on new version Windows like Windows 7, Windows 10.

I already know how to confirm system version, I have a boolean variable "is_xp_system".
Once the user run program on XP, that variable would be true, otherwise it should be false.
Convert output to local ANSI characters set if the variable is true... How to implement that?

**Comment by tako774** [comment]
nkf(1): Network Kanji Filter - Linux man page
https://linux.die.net/man/1/nkf

-s: output Shift_JIS
-x: keep halfwidth kana (such as 'ｱ') # default NKF converts halfwidth kana to fullwidth kana 'ア'
-m0: disable MIME decode # default NKF decode MIME encoded string

NKF (Network 'Kanji' Filter) supports only Japanese code pages (such as Shift-JIS, EUC-JP, ISO-2022-JP) and Unicode(UTF-8, UTF-16xx). Maybe iconv in ruby 1.8 standard library and google 先生 might help you (sry I have not used iconv with ruby 1.8).

**Comment by mhtvsSFrpHdE** [comment]
@tako774 So this is only work for Japanese and cp932.
There are many language on my development plan.
China, Taiwan, Hong Kong, Japan, USA, Korea languages.

Also I want make contributors easily to do and test translate.
I created a file here:
https://github.com/Incubator85653/tensokukan_network/blob/master/string.yaml

Any strings are stored in here and available to modify but without re-compile.
They will display correctly under UTF8 command shell.

But Windows DEFAULT cmd shell is not able to display the characters that not contains in local ANSI.
cp932, cp936, cp1252, etc.
To make tsk_report display any Unicode string, I found a third-party conhost interface can do this properly by "chcp 65001".

Windows XP conhost doesn't really have a 65001 mode.
That's why I need to convert stdout in XP even if that will get some string lost.

Thank you for your time.
I have to find a way that more generally to convert output encoding. For any language.
The best it could display non-Unicode characters as much as possible.
(Although convert Unicode to local ANSI will lost some string... better than display nothing.)
